sql/tables.go

```diff
@@ -166,6 +166,13 @@ type CheckAlterableTable interface {
 	DropCheck(ctx *Context, chName string) error
 }
 
+// CollationAlterableTable represents a table that supports the alteration of its collation.
+type CollationAlterableTable interface {
+	Table
+	// ModifyCollation modifies the collation that is set on the table.
+	ModifyCollation(ctx *Context, collation CollationID) error
```

Comment:
(minor/suggestion) I noticed there's also a "alter table <t> convert to character set" statement that will actually convert the table data to a different charset/collation, in addition to changing the default for the table. Assuming that'll go into this interface too (one day), you might consider naming this function `ModifyDefaultCollation` just to give it a little more space from a future `ConvertToCollation` function; alternatively, just mentioning that this function modifies the "default" collation in the godoc comment might help future readers (with less knowledge about collation details) more quickly understand what this function does.

Reply:
Makes sense to me! I'll change it